### PR TITLE
feat: API Visibility

### PR DESCRIPTION
**Pull request by shibeshduw**
DRAFT

Closes: [AI/gen-ai-hub-sdk-js#16](https://github.tools.sap/AI/gen-ai-hub-sdk-js-backlog/issues/16)

### CODE REVIEW

**Review by marikaner — CHANGES_REQUESTED** [review]
I am missing a script that checks the correctness of this or do we already have that? See: https://github.com/SAP/cloud-sdk-js/blob/main/scripts/check-public-api.ts and other files around that

File: packages/core/package.json [hunk 1]
```diff
@@ -14,13 +14,15 @@
...
+    "internal.js",
+    "internal.d.ts"
   ],
   "scripts": {
     "compile": "tsc",
     "test": "NODE_OPTIONS=--experimental-vm-modules jest",
-    "lint": "eslint . && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
-    "lint:fix": "eslint . --fix && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --log-level error"
+    "lint": "eslint \"**/*.ts\" && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
+    "lint:fix": "eslint \"**/*.ts\" --fix && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --log-level error"
```
[the remark this answers is not in the document]
Comment (shibeshduw) [reply packages/core/package.json:25]: Yes the `check-public-api.ts` was the script that had the typescript version dependency clash with the one in the cloud-sdk-js. Now that the version is updated, i will add the script back. 

I tried the `--ext` eslint option before pushing this and I believe this option is invalid for the current version of eslint we are using.

**Comment by marikaner** [comment]
@shibeshduw please add the check on a separate PR, I think this will be easier to review. Don't forget to adjust the description in here.

**Review by marikaner — DISMISSED** [review]
👍

**Review by marikaner — APPROVED** [review]

**Pull request merged**